### PR TITLE
NMS-12986: Fixed VMware polling and datacollection

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-12986

### CODE REVIEW

**Comment by RangerRick** [comment]
This looks like what I was imagining it would require, but tests are NPE'ing so I assume more changes are coming down the pipe.

**Comment by christianpape** [comment]
> This looks like what I was imagining it would require, but tests are NPE'ing so I assume more changes are coming down the pipe.

Test was missing the added TransactionTemplate bean. Fixed now.

**Comment by christianpape** [comment]
It seems, that the VMware Topology is also affected. I'll start to work on this...

File: features/topology-map/plugins/org.opennms.features.topology.plugins.topo.vmware/src/main/java/org/opennms/features/topology/plugins/topo/vmware/internal/VmwareTopologyProvider.java [hunk 1]
```diff
@@ -231,6 +232,8 @@ private AbstractVertex createHostSystemVertex(String vertexId, String vertexName
         final String vmwareTopologyInfo = getTopologyInfo(onmsNode);
 
         return Arrays.stream(vmwareTopologyInfo.split(SPLIT_REGEXP))
+                .map(s -> s.trim())
+                .filter(s -> !Strings.isNullOrEmpty(s))
```
Comment (fooker) [inline features/topology-map/plugins/org.opennms.features.topology.plugins.topo.vmware/src/main/java/org/opennms/features/topology/plugins/topo/vmware/internal/VmwareTopologyProvider.java:236]:
```suggestion
                .filter(s -> !s.isEmpty())
```
should be enough

File: integrations/opennms-vmware/src/main/java/org/opennms/netmgt/collectd/VmwareCimCollector.java [hunk 2]
```diff
@@ -193,6 +194,8 @@ private String applyUnitModifier(String attributeValue, String unitModifier) {
      */
     private NodeDao m_nodeDao = null;
 
+    private TransactionTemplate m_transactionTemplate = null;
```
Comment (fooker) [inline integrations/opennms-vmware/src/main/java/org/opennms/netmgt/collectd/VmwareCimCollector.java:197]: Use `SessionUtils` instead.

**Review by fooker — APPROVED** [review]

**Review by RangerRick — APPROVED** [review]

**Pull request merged**